apple/swift-docc-render
# Render combined navigation for merged documentation archives

Bug/issue #, if applicable: 124537023

## Summary

Now that Swift-DocC can merge multiple documentation archives together into a single archive, it will provide navigator data that includes a combined index for all the pages in the archive. This change updates Swift-DocC-Render to render this combined data for all the pages in the navigator.

Example:

<img width="1466" alt="Screenshot 2024-07-09 at 1 12 50 PM" src="https://github.com/swiftlang/swift-docc-render/assets/212918/5712abc7-9bf0-411c-b727-2b3c18e3f98c">

## Dependencies

https://github.com/apple/swift-docc/pull/821

## Testing

You can use this fixture which combines docs for `Inner` and `Outer` modules into a combined `My Project` archive:
[MyCombinedDocs.doccarchive.zip](https://github.com/user-attachments/files/16149848/MyCombinedDocs.doccarchive.zip)

Steps:
1. Download/unzip the example `MyCombinedDocs.doccarchive.zip` fixture and checkout this branch of Swift-DocC-Render
2. Run `VUE_APP_DEV_SERVER_PROXY=/path/to/MyCombinedDocs.doccarchive npm run serve` and open http://localhost:8080/documentation
3. Verify that the top-level page for "My Project" loads at that URL and that the navigator has functional links to both the `Inner` and `Outer` docs
4. Verify that navigation works normally and there are no regressions with routing behavior

Known DocC issue: the breadcrumb link to "My Project" from the getting started article doesn't work due to a bug with the compiler data pointing to the wrong URL (should be /documentation and not /documentation/my-project)

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
